### PR TITLE
FORG-32 Roll over system

### PR DESCRIPTION
**Pull request by AndrewMikhaylov**

### CODE REVIEW

**Comment by Marichellka** [comment]
please add ticket number to PR name

File: Frogies-Project/Assets/Scripts/Animation/PlayerAnimationState.cs [hunk 1]
```diff
@@ -7,12 +7,13 @@ public enum PlayerAnimationState
...
         Idle = 0,
         Run = 1,
-        Attack = 2,
-        Attack2 = 3,
-        QuickTurn = 4,
-        Jump = 5,
-        Fall = 6,
-        GetHit = 7,
-        Death = 8,
+        RollOver = 2,
```
Comment (Marichellka) [inline Frogies-Project/Assets/Scripts/Animation/PlayerAnimationState.cs:10]: please remove index as they go by default

File: Frogies-Project/Assets/Prefabs/Scene/BasicScenePrefab.prefab
Comment (Marichellka) [inline Frogies-Project/Assets/Prefabs/Scene/BasicScenePrefab.prefab:1]: please update prefab with new items

File: Frogies-Project/Assets/Scripts/Core/GlobalSceneManager.cs [hunk 2]
```diff
@@ -36,7 +36,7 @@ public class GlobalSceneManager : MonoBehaviour
         [SerializeField] private Inventory inventory;
         [SerializeField] private WaveStorage _waveStorage;
 
-        [SerializeField] private PlayerData playerData;
+        [SerializeField] public PlayerData playerData;
```
Comment (Marichellka) [inline Frogies-Project/Assets/Scripts/Core/GlobalSceneManager.cs:39]: Why do we need it public? (Haven't seen where it used)
Can you somehow get it from player instead?

File: Frogies-Project/Assets/Scripts/Core/GlobalSceneManager.cs [hunk 3]
```diff
@@ -78,14 +78,14 @@ private void Awake()
...
 
-        private void InitializeItemFactory(BasicEntity player)
-        {
-            ItemFactory factory = new ItemFactory(player.Brain.StatsController);
-            _sceneItemStorage = new ItemSystem(
-                PrefabsStorage.SceneItemPrefab.GetComponent<SceneItem>(), 
-                itemRarityDescriptor.RarityDescriptor.Cast<IItemRarityColor>().ToArray(), 
-                factory, inventory);
-        }
+         private void InitializeItemFactory(BasicEntity player)
```
Comment (Marichellka) [inline Frogies-Project/Assets/Scripts/Core/GlobalSceneManager.cs:81]: ????
please revert these changes

File: Frogies-Project/Assets/Scripts/Core/GlobalSceneManager.cs [hunk 4]
```diff
@@ -117,14 +117,14 @@ public BasicEntity InitializeEnemy(GameObject enemyPrefab, out GameObject enemyG
...
 
-        private void InitializePotionSystem(List<ItemDescriptor> itemDescriptors, BasicEntity player)
-        {
-            var depowerPotions = itemDescriptors.Where(descriptor => descriptor.ItemId == ItemId.DepowerPotion)
-                .Select(descriptor => new Potion(descriptor as StatChangingItemDescriptor, player.Brain.StatsController)).ToList();
-            potionSystem.Setup(depowerPotions);
-            potionSystem.OnActive += () => _isPaused = true;
-            potionSystem.OnOptionSelected += _ => _isPaused = false;
-        }
+         private void InitializePotionSystem(List<ItemDescriptor> itemDescriptors, BasicEntity player)
```
Comment (Marichellka) [inline Frogies-Project/Assets/Scripts/Core/GlobalSceneManager.cs:120]: same here

File: Frogies-Project/Assets/Scripts/Fighting/DamageReceiver.cs [hunk 5]
```diff
@@ -14,6 +15,10 @@ public void Initialize(Action<float> onDamageReceived)
 
         public void ReceiveDamage(float damage)
         {
+            // if (GlobalSceneManager.Instance.playerData.DirectionalMover.IsDashing)
```
Comment (Marichellka) [inline Frogies-Project/Assets/Scripts/Fighting/DamageReceiver.cs:18]: why do we need it commented? remove / uncomment or add comment why it's needed

**Comment by Marichellka** [comment]
feature isn't working at all

**Review by Sorpirit — CHANGES_REQUESTED** [review]
Also make sure that roll over auto cancels when player hits a wall

**Review by vakor03 — APPROVED** [review]

**Review by Sorpirit — APPROVED** [review]

**Pull request merged**